### DISCUSSION TITLE
Allow user to add a value to numeric entries

### DISCUSSION THREAD

**Discussion by Stiralbios** [iSoron/uhabits#951] [Feature Requests]
On long press on a numerical habbit we can edit it. It would be nice to have an option in the popup to add a value to the previous registered value.

For exemple if you read a book in differents moment of the day and you read 8 pages then 14 then 9. You have to do the calculation in you head when you edit the value. This option could make things easier for the user.

**Comment by ke352802081770314** [comment]
This was also one of my thoughts when getting started with this app. (Edit: sorry, I only just spotted the small up-arrow method of supporting a request, apologies for creating noise by answering.)

**Comment by hiqua** [comment]
Maybe a double-press (rather than quick single press / long single press, depending on the settings) could trigger a pop-up prompting the user for the increase.

It could also make sense to allow negative amounts to decrease the value.

**Reply by ke352802081770314** [reply]
I think different press methods are not intuitive. I would understand these for something that gives immediate feedback and is easy to revert. Something that opens a dialogue is complex, and having one more selection is less hassle than having to understand why you ended up with the dialogie you did, plus people already using the app will have to know of the added functionality instead of experiencing it.
I see two options that in my opinion could make sense:
A - have one dialogue, like now, but with two submit/confirm buttons, one for adding and one for setting. The add button need only appear if a value was already set previously. We assume that add is the morecomon use-case, so we have to start at 0, but for better understanding of the situation the current value should be displayed as a piece of information.
B - first have a dialogue asking what to do: add or replace value, then go to the entry dialogue. The first dialogue can be skipped if no entry has yet been made in the day. The entry dialogue will be pre-filled with the current value for "replace/change/set" and pre-filled with 0 for "add". Displaying the current value is not really needed in that case.
Please refer to the following mockup screenshots for illustration (and sorry I am not a designer so they are ugly, this is only about content, not style).

![20210605-loop-add_set](https://user-images.githubusercontent.com/10772063/120893303-e34b3580-c612-11eb-944c-84b1537103fd.png)

Personally I would prefer the two-step approach. Two versions of submit button results in higher risk of mistyping. Actually, I think the "numerical" would benefit from developing into something more diverse, I will elaborate on this in a separate thread.

PS: forgot to say: I think negatives are not that helpful. In all use cases I can think of, you record actions as they happen, and you would only use negatives for correcting a mis-entry, correcting this is likely to be better informed with the total in mind so by setting the total to the true value instead of adding a negative difference.

**Reply by hiqua** [reply]
> Two versions of submit button results in higher risk of mistyping. 

I think the problem could be confusion and hitting the wrong button thinking it means the other, rather than mistyping because of a fat finger.

Given this
1. for A., the "add" button could just be way smaller than the "set" button. This makes sense if we assume that users mostly "set" the value rather than "add" to it. For users that routinely "add", they'll probably do it enough to get used to the smaller button anyway.

2. or simply combine B(1) and B(2) in one screen, so that the options in B(1) is just a toggle. when "add" is selected, also display the resulting updated value. For every habit, remember what position the toggle was for the last entry, and use it next time (I suspect most habits will be "add" only or "set" only).

I don't think we want to add friction to enter habits, numerical habits are already significantly more complex than basic ones, which does not help users tracking them. So having two different screen is I think worse than 2.

Regarding negative increments, I actually thought signed values were supported, but that's actually not the case. Having default values would also make it useful, but that's also not currently supported. So I agree with you that it's not relevant for now, maybe later.
